src/main/java/com/github/orest/car_shop/controllers/CarController.java

```diff
@@ -0,0 +1,114 @@
+package com.github.orest.car_shop.controllers;
+
+import com.github.orest.car_shop.model.Car;
+import com.github.orest.car_shop.repository.CarRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.servlet.view.RedirectView;
+
+import javax.validation.Valid;
+import java.util.Map;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+@RequestMapping
```

Comment:
You should have created REST endpoints, please read these articles:

1. https://spring.io/guides/tutorials/rest/
2. https://habr.com/ru/post/351890/
3. https://habr.com/ru/post/38730/

Reply:
will add a separate CarApiController for rest endpoints